waddrmgr/manager_test.go

```diff
@@ -1019,38 +1019,64 @@ func testImportScript(tc *testContext) bool {
 
 // testMarkUsed ensures used addresses are flagged as such.
 func testMarkUsed(tc *testContext) bool {
-	expectedAddr1 := expectedAddr{
-		addressHash: hexToBytes("2ef94abb9ee8f785d087c3ec8d6ee467e92d0d0a"),
-		used:        true,
+	tests := []struct {
+		name       string
+		addrType   btcutil.Address
```

Comment:
The type itself is used, but not the data.  I would consider using constants here instead:

func testMarkUsed(tc *testContext) bool {
    type addrType byte
    const (
        addrPubKeyHash addrType = iota
        addrScriptHash
    )
    tests := []struct {
        // ...
        typ addrType
        // ...
    }

    // ...

        switch test.typ {
        case addrPubKeyHash:
            // ...
        case addrScriptHash:
            // ...
        default:
            panic("unreachable")
        }

    // ...
}
```